flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -451,12 +451,23 @@ private void setupDispatcherResourceManagerComponents(
             final CompletableFuture<ApplicationStatus> shutDownFuture =
                     dispatcherResourceManagerComponent.getShutDownFuture();
             FutureUtils.assertNoException(
-                    shutDownFuture.thenRun(dispatcherResourceManagerComponent::closeAsync));
+                    shutDownFuture
+                            .handle(
+                                    (applicationStatus, exception) -> {
+                                        if (exception != null) {
+                                            return dispatcherResourceManagerComponent
+                                                    .stopApplication(
+                                                            ApplicationStatus.UNKNOWN,
+                                                            ExceptionUtils.stringifyException(
+                                                                    exception));
```

Comment:
oh, this is a leftover from the previous approach. I think we currently never complete shutdownFuture exceptionally, so we can safely ignore this and let the `assertNoException` take care of it in case it happens 👍